lib/util/movingaveragedelta.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+/**
+ * Calculates the moving average delta for the given samples. A sample (S) consists of
+ * a numerator (Sn) and a denominator (Sd). Given the sample size n, the moving average
+ * delta is calculated as follows:
+ *
+ * AvgDelta = Sum((Sn[i] - Sn[i-1]) / (Sd[i] - Sd[i-1])) for all i in [1, n - 1]
+ */
+class MovingAverageDelta {
+  /**
+   * Constructor.
+   * @param size
+   */
+  constructor(size) {
+    Object.defineProperties(this, {
+      _samples: {
+        value: [],
+      },
+      _size: {
+        value: size
+      }
+    });
+  }
+
+  /**
+   * Get the moving average delta.
+   * @returns {number}
+   */
+  get() {
+    const { _samples: samples } = this;
+    if (samples.length <= 0) {
+      return 0;
+    }
+    if (samples.length === 1) {
+      return Math.round(samples[0].numerator / samples[0].denominator);
+    }
+    const sampleDifferencesRight = samples.slice(0, samples.length - 1);
```

Comment:
I did not understand this math -  some comments would be useful here.

I did not understand why do you need to keep 3 samples. Should we just keep last sample? and compute averages using
(current Numerator - last Numerator) / (current denominator - last denominator)